blackbox_repository/conversion_scripts/scripts/fcnet_import.py

```diff
@@ -0,0 +1,161 @@
+from blackbox_repository.conversion_scripts.utils import repository_path, upload
+from syne_tune.util import catchtime
+
+
+def convert_dataset(dataset_path: Path, max_rows: int = None):
+    data = h5py.File(dataset_path, "r")
+    keys = data.keys()
+    if max_rows is not None:
+        keys = list(keys)[:max_rows]
+
+    hyperparameters = pd.DataFrame(ast.literal_eval(key) for key in keys)
+    hyperparameters.rename(columns={col: "hp_" + col for col in hyperparameters.columns}, inplace=True)
+
+    objective_names = [
+        'valid_loss',
+        'train_loss',
+        # only use train/valid loss as mse are the same numbers
+        # 'valid_mse',
+        # 'train_mse',
+        'final_test_error',
+        'n_params',
+        'runtime',
+    ]
+
+    # todo for now only full metrics
+    fidelity_values = np.arange(1, 101)
+    n_fidelities = len(fidelity_values)
+    n_objectives = len(objective_names)
+    n_seeds = 4
+    n_hps = len(keys)
+
+    objective_evaluations = np.empty((n_hps, n_seeds, n_fidelities, n_objectives)).astype('float32')
+
+    def save_objective_values_helper(name, values):
+        assert values.shape == (n_hps, n_seeds, n_fidelities)
+
+        name_index = dict(zip(
+            objective_names,
+            range(len(objective_names)))
+        )
+        objective_evaluations[..., name_index[name]] = values
+
+    # (n_hps, n_seeds,)
+    final_test_error = np.stack([data[key]['final_test_error'][:].astype('float32') for key in tqdm(keys)])
+
+    # (n_hps, n_seeds, n_fidelities)
+    final_test_error = np.repeat(np.expand_dims(final_test_error, axis=-1), n_fidelities, axis=-1)
+    save_objective_values_helper('final_test_error', final_test_error)
+
+    # (n_hps, n_seeds,)
+    n_params = np.stack([data[key]['n_params'][:].astype('float32') for key in tqdm(keys)])
+
+    # (n_hps, n_seeds, n_fidelities)
+    n_params = np.repeat(np.expand_dims(n_params, axis=-1), n_fidelities, axis=-1)
+    save_objective_values_helper('n_params', n_params)
+
+    # (n_hps, n_seeds,)
+    runtime = np.stack([data[key]['runtime'][:].astype('float32') for key in tqdm(keys)])
+
+    # linear interpolation to go from total training time to training time per epoch as in fcnet code
+    # (n_hps, n_seeds, n_epochs)
+    # todo utilize expand dim instead of reshape
+    epochs = np.repeat(np.arange(1, 101).reshape(1, -1), n_hps * n_seeds, axis=0).reshape(n_hps, n_seeds, -1)
+    runtime_per_epoch = (100 / epochs) * runtime.reshape((n_hps, n_seeds, 1))
+    
+    save_objective_values_helper('runtime', runtime_per_epoch)
+
+    # metrics that are fully observed, only use train/valid loss as mse are the same numbers
+    # for m in ['train_loss', 'train_mse', 'valid_loss', 'valid_mse']:
+    for m in ['train_loss', 'valid_loss']:
+        save_objective_values_helper(
+            m,
+            np.stack([data[key][m][:].astype('float32') for key in tqdm(keys)])
+        )
+
+    configuration_space = {
```

Comment:
The blackbox does not necessarily requires a surrogate interpolation model, it depends. 
* In (tabular) cases where all possibilities are acquired such as FCNET or NAS201 for instance, then one can just sample from the search-space and the candidate will be there.
* In cases where only partial observations are observed from the search-space (for instance if data is acquired with lograndint(8, 64) or loguniform for a learning-rate), then a surrogate must be used since a sampled configuration may not be present

For instance in the examples provided in launch_simulated_benchmark.py the first 2 examples does not use a surrogate (all possibilities are there) but the third one uses a surrogate since only partial evaluations are present.

For a tabular benchmark, one could use a different search-space if a surrogate model is used.

Reply:
Regarding the config-space used in the current blackbox, the idea is to take the search-space as defined by the author (but other search-space could be used as indicated in my previous comment if interpolation is activated).